### PR TITLE
Glossary of terms

### LINKED ISSUE #63: Glossary is incomplete
https://github.com/OBOAcademy/obook/blob/master/docs/reference/glossary.md

An OBO-centric glossary of concepts would be incredibly useful.

 - Should we crowd source this a bit more? I tend to think that it should be a few people to stay coherent
 - Will it become hard to maintain, like a large ontology? 
 - What should the scope be? Should it for example include all concepts in OWL (restriction, disjointness, QCR, GCI, TBox, ABox)? Or should these be in a specialized glossaries?
 - should we consider a full blown skos and a vocabulary management tool? Or a wikidata/wikipedia layer?

### PR DESCRIPTION
**Pull request by shawntanzk**
Fixes https://github.com/OBOAcademy/obook/issues/210
Fixes #63

### CODE REVIEW

**Comment by shawntanzk** [comment]
Heaps of spots that are empty now that need filling - will need to get through it bit by bit. After this round of terms (once we complete them) I think we can do a first release of this. From there we can edit and add as we want

**Comment by shawntanzk** [comment]
Things to consider -> should we arrange them in some way? would take a bit of work to get them alphabetical, but not impossible.

**Comment by shawntanzk** [comment]
Did what I can for now, needs proper review before going out.

File: docs/reference/glossary-of-terms.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
+# Glossary of Terms
+
+This document is a list of terms that you might encounter in the ontology world. It is not an exhaustive list and will continue grow and evolve. Please do create a ticket if there is a term you find missing here, or a term that you encounter that you do not understand, and we will do our best to add them in here. This list is not arranged in any particular order, please use the search function to find needed term.
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:3]: Minor edits:
This document is a list of terms that you might encounter in the ontology world. It is not an exhaustive list and will continue to evolve. Please create a ticket if there is a term you find missing or a term you encounter that you do not understand, and we will do our best to add them. This list is not arranged in any particular order. Please use the search function to find needed term.

Comment (bvarner-ebi) [reply docs/reference/glossary-of-terms.md:3]: > This list is not arranged in any particular order

Consider alphabetising.

File: docs/reference/glossary-of-terms.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+Acknowledgement: Many terms are taken directly from [OAK documentation](https://incatools.github.io/ontology-access-kit/glossary.html) with the permission of Chris Mungall. Many descriptions are also taken from https://www.w3.org/TR/owl2-syntax/
+
+#### Ontology
+A flexible concept loosely encompassing any collection of [Ontology elements](#Ontology-element) and statements or relationships connecting them.
+
+#### Knowledge Graph
+A network of real-world entities—i.e. objects, events, situations, or concepts—and illustrates the relationship between them. Knowledge graphs (in the ontologies world) are thought of as real data built using an ontology as a framework.
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:11]: Minor edits:
A network of real-world entities (i.e., objects, events, situations, and concepts) that illustrates the relationships between them.

Comment (bvarner-ebi) [reply docs/reference/glossary-of-terms.md:11]: > Knowledge graphs (in the ontologies world) are thought of as real data built using an ontology as a framework.

The second use of the term "ontology world"... what is an "ontology world"? Are there knowledge graphs that have a different meaning that exist outside of this "world"?

File: docs/reference/glossary-of-terms.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+#### Controlled Vocabulary
+A standardized and organized arrangements of words and phrases and provide a consistent way to describe data. A controlled vocabulary may have no meaning specified (just a set of terms people agree on), or may also have detailed definitions. Ontologies can be seen as a controlled vocabulary expressed in an ontological language which includes relations.
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:14]: Minor edits:
Standardized and organized arrangements of words and phrases that provide a consistent way to describe data. A controlled vocabulary may or may not include definitions. Ontologies can be seen as a controlled vocabulary expressed in an ontological language which includes relations.

File: docs/reference/glossary-of-terms.md (hunk 1 above)
Comment (nicolevasilevsky) [reply docs/reference/glossary-of-terms.md:3]: I can alphabetize this list. I'll do so on this branch.

Comment (nicolevasilevsky) [reply docs/reference/glossary-of-terms.md:3]: I revised the text per @bvarner-ebi's suggestion

File: docs/reference/glossary-of-terms.md (hunk 2 above)
Comment (nicolevasilevsky) [reply docs/reference/glossary-of-terms.md:11]: I made this revision:
A network of real-world entities (i.e., objects, events, situations, and concepts) that illustrates the relationships between them.

File: docs/reference/glossary-of-terms.md (hunk 3 above)
Comment (nicolevasilevsky) [reply docs/reference/glossary-of-terms.md:14]: I made this revision

File: docs/reference/glossary-of-terms.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+#### Class
+An [Ontology elements](#Ontology-element) that formally represents something that can be instantiated. For example, the class "heart"
+
+#### CURIE
+A CURIE is a compact [URI](#URI). For example, `CL:0000001`.
+
+#### Data Model
+A representation of the structure and integrity of the data elements of the, in principle "single", specific enterprise application(s) by which it will be used.
+
+#### Datamodel
```
Comment (nicolevasilevsky) [inline docs/reference/glossary-of-terms.md:46]: is this different than above?

File: docs/reference/glossary-of-terms.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+#### DatatypeProperty
+DatatypeProperty relate terms to literal data (e.g., strings, numbers, datetimes, etc.) as opposed to [ObjectProperty](#ObjectProperty) which relate individuals to other terms. 
+
+#### Description Logic
+Description Logic (DL) are a family of formal knowledge representation languages. It provides a logical formalism for ontologies and is what [OWL](#OWL) is based on. DL querying can be used to query ontologies in Protege.
+
+#### Edge
+[ObjectProperty](#ObjectProperty) (or relations) represented in a [graph](#Graph)
+
+#### Edge
```
Comment (nicolevasilevsky) [inline docs/reference/glossary-of-terms.md:58]: is this different than above?

File: docs/reference/glossary-of-terms.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+A core element in an ontology, typically a [Class](#Class), but sometimes used to include instances or relationship types, depending on context.
+
+#### Text Annotation
+The process of annotating spans of texts within a text document with references to ontology terms, or the result of this process. This is frequently done automatically. The [Bioportal](#Bioportal) implementation provides text annotation services.
+
+#### Triple
+A set of three entities that codifies a statement about semantic data in the form of [Subject](#Subject)-[predicate](#predicate)-[Object](#Object) expressions (e.g., "Bob is 35", or "Bob knows John").
+
+#### Triple
+See [Relationship](#Relationship)
```
Comment (nicolevasilevsky) [inline docs/reference/glossary-of-terms.md:206]: is this different than above?

**Comment by shawntanzk** [comment]
omg thank you so much @nicolevasilevsky! I'll go through the stuff - I think some might have been repeated cause migration from OAK docs was very manual and I was trying to fill in gaps from there

**Comment by nicolevasilevsky** [comment]
Sure thing @shawntanzk! Thank you for creating this, it will be so helpful.

File: docs/reference/glossary-of-terms.md [hunk 7]
```diff
@@ -49,6 +49,9 @@ DatatypeProperty relate terms to literal data (e.g., strings, numbers, datetimes
 #### Description Logic
 Description Logic (DL) are a family of formal knowledge representation languages. It provides a logical formalism for ontologies and is what [OWL](#OWL) is based on. DL querying can be used to query ontologies in Protege.
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:50]: I believe "Description Logic**s**" fits the definition provided. Description Logic (singular) would be one of the varieties from the "family" described (Attributive language, Existential language, etc). This is my understanding, but @henrietteharmse would have better insight.

File: docs/reference/glossary-of-terms.md [hunk 8]
```diff
@@ -49,6 +49,9 @@ DatatypeProperty relate terms to literal data (e.g., strings, numbers, datetimes
...
 
+#### DOSDP
+[Dead Simple Ontology Design Patterns](http://incatools.github.io/dead_simple_owl_design_patterns/). A templating system for ontologies with well-documented patterns and templates.
+
 #### Edge
 [ObjectProperty](#ObjectProperty) (or relations) represented in a [graph](#Graph). Also see See [Relationship](#Relationship)
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:56]: "Also see See ..."

File: docs/reference/glossary-of-terms.md [hunk 9]
```diff
@@ -85,6 +88,9 @@ A network of real-world entities (i.e., objects, events, situations, and concept
 #### Label
 Usually refers to a human-readable label corresponding to the `rdfs:label` [predicate](#predicate). Labels are typically unique per ontology. In OBO Format and in the bio-ontology literature, labels are sometimes called [Names](#Name). Sometimes in the machine learning literature, and in databases such as Neo4J, "label" actually refers to a [Category](#Category).
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:89]: Minor edits:
Usually refers to a human-readable text string corresponding to the `rdfs:label` [predicate](#predicate). Labels are typically unique per ontology. In OBO Format and in the bio-ontology literature, labels are sometimes called [Names](#Name). Sometimes in the machine learning literature and in databases such as Neo4J, "label" actually refers to a [Category](#Category).

File: docs/reference/glossary-of-terms.md [hunk 10]
```diff
@@ -85,6 +88,9 @@ A network of real-world entities (i.e., objects, events, situations, and concept
...
 
+#### Lutra
+[Lutra](https://ottr.xyz/#Lutra) is the open source reference implementation of the OTTR templating language.
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:92]: Can OTTR be linked to the OTTR definition on this page?

File: docs/reference/glossary-of-terms.md [hunk 11]
```diff
@@ -85,6 +88,9 @@ A network of real-world entities (i.e., objects, events, situations, and concept
...
+
 #### Mapping
 See [SSSOM]('SSSOM')
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:95]: SSSOM term on this page does not provide a definition for "Mapping".

File: docs/reference/glossary-of-terms.md [hunk 12]
```diff
@@ -127,12 +136,18 @@ The systems or platform where various types of ontologies are stored from differ
 #### Ontology Repository 
 A curated collection of ontologies.
 
+### OTTR
+[Reasonable Ontology Templates](https://ottr.xyz/). A system for composable ontology templates and documentation.
+
 #### OWL
 An ontology language that uses constructs from [Description Logic](#Description-Logic). OWL is not itself an ontology format, it can be serialized through different formats such as [Functional Syntax](#Functional-Syntax), and it can be mapped to :[RDF](#RDF) and serialized via an RDF format.
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:143]: Consider adding what this stands for (i.e., Web Ontology Language with the letters reorganised for easier articulation).

File: docs/reference/glossary-of-terms.md [hunk 13]
```diff
@@ -127,12 +136,18 @@ The systems or platform where various types of ontologies are stored from differ
...
 
 #### OWL Annotation
 In the context of [OWL](#OWL), the term [Annotation](#Annotation) means a piece of metadata that does not have a strict logical interpretation. Annotations can be on entities, for example, [Label](#Label) annotations, or annotations can be on [Axioms](#Axiom).
 
+#### OWL API
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:148]: Consider adding what API stands for (i.e., application programming interface).

File: docs/reference/glossary-of-terms.md [hunk 14]
```diff
@@ -127,12 +136,18 @@ The systems or platform where various types of ontologies are stored from differ
...
+A java-based API to interact with OWL ontologies. Full documentation can be found at http://owlcs.github.io/owlapi/apidocs_5/index.html
+
 #### Predicate
 An [Ontology elements](#Ontology-element) that represents the type of a [Relationship](#Relationship).
```
Comment (bvarner-ebi) [inline docs/reference/glossary-of-terms.md:152]: "An [Ontology elements]..." --> "An [Ontology element]..."

File: docs/reference/glossary-of-terms.md [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+
+This document is a list of terms that you might encounter in the ontology world. It is not an exhaustive list and will continue to evolve. Please create a ticket if there is a term you find missing or a term you encounter that you do not understand, and we will do our best to add them. This list is not arranged in any particular order. Please use the search function to find terms.
+
+Acknowledgement: Many terms are taken directly from [OAK documentation](https://incatools.github.io/ontology-access-kit/glossary.html) with the permission of Chris Mungall. Many descriptions are also taken from https://www.w3.org/TR/owl2-syntax/
+
+#### Annotation
+This term is frequently ambiguous. It can refer to [Text Annotation](#Text-Annotation), [OWL Annotation](#OWL Annotation), or [Association](#Association).
+
+#### AnnotationProperty
+Annotation properties are used to place annotations on individuals, class names, property names, and ontology names. They do not affect the logical definition unless they are used as a "shortcut" that a pipeline expands to a logical axiom.
```
Comment (dosumis) [inline docs/reference/glossary-of-terms.md:11]: Make clear this is OWL specific

File: docs/reference/glossary-of-terms.md [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+An [Ontology elements](#Ontology-element) that formally represents something that can be instantiated. For example, the class "heart"
+
+#### CURIE
+A CURIE is a compact [URI](#URI). For example, `CL:0000001`.
+
+#### Data Model
+An abstract model that organizes elements of data and standardizes how they relate to one another and to the properties of real-world entities.
+
+#### DatatypeProperty
+DatatypeProperty relate terms to literal data (e.g., strings, numbers, datetimes, etc.) as opposed to [ObjectProperty](#ObjectProperty) which relate individuals to other terms.
```
Comment (dosumis) [inline docs/reference/glossary-of-terms.md:47]: Doesn't sufficiently differentiate from AnnotationProperty which can also relate terms (better to say Ontology element or OWL Entity?) to strings, number, datatimes, etc).  The difference is that Data  Property axoims are on the logical side of OWL - usable by reasoners.  Annotation property axioms are not.

File: docs/reference/glossary-of-terms.md (hunk 11 above)
Comment (shawntanzk) [reply docs/reference/glossary-of-terms.md:95]: lol copied over from OAK - told myself I needed to fix this but forgot

File: docs/reference/glossary-of-terms.md [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+DatatypeProperty relate terms to literal data (e.g., strings, numbers, datetimes, etc.) as opposed to [ObjectProperty](#ObjectProperty) which relate individuals to other terms. 
+
+#### Description Logic
+Description Logic (DL) are a family of formal knowledge representation languages. It provides a logical formalism for ontologies and is what [OWL](#OWL) is based on. DL querying can be used to query ontologies in Protege.
+
+#### DOSDP
+[Dead Simple Ontology Design Patterns](http://incatools.github.io/dead_simple_owl_design_patterns/). A templating system for ontologies with well-documented patterns and templates.
+
+#### Edge
+[ObjectProperty](#ObjectProperty) (or relations) represented in a [graph](#Graph). Also see See [Relationship](#Relationship)
```
Comment (dosumis) [inline docs/reference/glossary-of-terms.md]: Need to distinguish between objectProperty and an axiom containing an object property.  I'd define as 
"A typed, directed link between nodes in a [knowledge graph]().  Translations of OWL into Knowledge graphs vary, but typically edges are generated for simple triples (relating two individuals or two classes via an ObjectProperty or AnnotationProperty) and simple existential restrictions (A SubClassOf R some B), with the edge type corresponding to the property.

File: docs/reference/glossary-of-terms.md [hunk 18]
```diff
@@ -0,0 +1,227 @@
...
+
+#### Nodes
+[Terms](#Term) represented in a [graph](#Graph)
+
+#### Object
+The "right" side of a [Triple](#Triple).
+
+#### ObjectProperty
+A simple association between two terms, also see [predicate](#predicate) & [Relationship](#Relationship). 
+
```
Comment (dosumis) [inline docs/reference/glossary-of-terms.md:114]: I think this needs work.  I'm really not sure about the loose use of term throughout this glossary.  I think easier to use OWL terminology of classes and individuals etc & have an entry for term explaining its general usage.

File: docs/reference/glossary-of-terms.md [hunk 19]
```diff
@@ -0,0 +1,227 @@
...
+#### Ontology
+A flexible concept loosely encompassing any collection of [Ontology elements](#Ontology-element) and statements or relationships connecting them.
+
+#### ODK
+[Ontology Development Kit](https://github.com/INCATools/ontology-development-kit). A toolkit and docker image for managing ontologies.
+
+#### Ontology element
+A discrete part of an [Ontology](#Ontology), with a unique persistent identifier. The most important elements are [Terms](#Term) but other elements can include various metadata artefacts like [Annotation Properties](#Annotation-Property) or [Subsets](#Subset)
+
```
Comment (dosumis) [inline docs/reference/glossary-of-terms.md:132]: Term is confusing here.  Elsewhere Ontology element seems to correspond to OWL Entity.  Might be better to make that clearer.  Also odd to see Subset here.

File: docs/reference/glossary-of-terms.md [hunk 20]
```diff
@@ -0,0 +1,227 @@
...
+### OTTR
+[Reasonable Ontology Templates](https://ottr.xyz/). A system for composable ontology templates and documentation.
+
+#### OWL
+An ontology language that uses constructs from [Description Logic](#Description-Logic). OWL is not itself an ontology format, it can be serialized through different formats such as [Functional Syntax](#Functional-Syntax), and it can be mapped to :[RDF](#RDF) and serialized via an RDF format.
+
+#### OWL Annotation
+In the context of [OWL](#OWL), the term [Annotation](#Annotation) means a piece of metadata that does not have a strict logical interpretation. Annotations can be on entities, for example, [Label](#Label) annotations, or annotations can be on [Axioms](#Axiom).
+
```
Comment (dosumis) [inline docs/reference/glossary-of-terms.md:147]: I think worth explicitly refering to annotation axioms.

File: docs/reference/glossary-of-terms.md (hunk 7 above)
Comment (shawntanzk) [reply docs/reference/glossary-of-terms.md:50]: logics is write for the description :)

**Review by dosumis — CHANGES_REQUESTED** [review]
I've added a mix of specific and general comments.  In general I think the OWL side needs to be described in OWL terms (avoid talking about relationships, edges, terms, ontology elements in*in* OWL) - instead using terms like axiom, Class, OWLEntity.  We absolutely need entries for relationship, edge, term etc in the glossary.   These should be partly defined by their (often loose) mapping to OWL.

**Comment by shawntanzk** [comment]
ontology elements should = owl entity 
property vs axiom -> need to differentiate (eg OP does not = relations, but axioms that use OP = relations) 

Use case: link from main text 

Fromat -> possibly not alphabetical 

need clear distinction between OWL terminology vs OBO-esque stuff

File: docs/reference/glossary-of-terms.md [hunk 21]
```diff
@@ -0,0 +1,227 @@
+# Glossary of Terms
+
+This document is a list of terms that you might encounter in the ontology world. It is not an exhaustive list and will continue to evolve. Please create a ticket if there is a term you find missing or a term you encounter that you do not understand, and we will do our best to add them. This list is not arranged in any particular order. Please use the search function to find terms.
+
+Acknowledgement: Many terms are taken directly from [OAK documentation](https://incatools.github.io/ontology-access-kit/glossary.html) with the permission of Chris Mungall. Many descriptions are also taken from https://www.w3.org/TR/owl2-syntax/.
```
Comment (nicolevasilevsky) [inline docs/reference/glossary-of-terms.md:5]:
```suggestion
Acknowledgement: Many terms are taken directly from [OAK documentation](https://incatools.github.io/ontology-access-kit/glossary.html) with the permission of [Chris Mungall](https://orcid.org/0000-0002-6601-2165). Many descriptions are also taken from https://www.w3.org/TR/owl2-syntax/.
```

File: docs/reference/glossary-of-terms.md [hunk 22]
```diff
@@ -0,0 +1,227 @@
+# Glossary of Terms
```
Comment (nicolevasilevsky) [inline docs/reference/glossary-of-terms.md:1]: Call this 'glossary instead'

Comment (nicolevasilevsky) [inline docs/reference/glossary-of-terms.md:1]:
```suggestion
# Glossary
```

File: docs/reference/glossary-of-terms.md [hunk 23]
```diff
@@ -0,0 +1,300 @@
...
+A set of three entities that codifies a statement about semantic data in the form of [Subject](#Subject)-[predicate](#predicate)-[Object](#Object) expressions (e.g., "Bob is 35", or "Bob knows John"). Also see [Relationship](#Relationship).
+
+#### Triplestore
+
+A purpose-built database for the storage and retrieval of triples through semantic queries. A triple is a data entity composed of subject–predicate–object, like "Bob is 35" or "Bob knows Fred".
+
+#### Ubergraph
+
+A [Triplestore](#Triplestore) and a [Ontology Repository](#Ontology-Repository) that allows for [SPARQL](#SPARQL) querying of integrated [OBO](#OBO) ontologies.
+
```
Comment (dosumis) [inline docs/reference/glossary-of-terms.md:297]: Kinda understates what Ubergraph is and why it's useful.  Link to original and maybe take some text from their readme.

**Review by dosumis — CHANGES_REQUESTED** [review]

File: docs/reference/glossary-of-terms.md [hunk 24]
```diff
@@ -0,0 +1,300 @@
...
+Standardized and organized arrangements of words and phrases that provide a consistent way to describe data. A controlled vocabulary may or may not include definitions. Ontologies can be seen as a controlled vocabulary expressed in an ontological language which includes relations.
+
+#### Class
+
+An [OWL entity](#OWL-entity) that formally represents something that can be instantiated. For example, the class "heart".
+
+#### CURIE
+
+A CURIE is a compact [URI](#URI). For example, `CL:0000001`.
```
[the remark this answers is not in the document]
Comment (balhoff) [reply docs/reference/glossary-of-terms.md:49]: Link to latest: https://www.w3.org/TR/curie/

**Comment by shawntanzk** [comment]
Merging now given the last requested changes are made - iterate if any other issues are spotted

**Pull request merged**